src/com/owncloud/android/datamodel/OCFile.java

```diff
@@ -747,4 +749,36 @@ public void copyLocalPropertiesFrom(OCFile sourceFile) {
         setEtagInConflict(sourceFile.getEtagInConflict());
     }
 
+    /**
+     * Manipulate file remote id to get private link
+     *
+     * @param context
+     * @param account
+     * @return private link for the current file
+     */
+    public String getPrivateLink(Context context, Account account) {
+
+        // Parse remoteId
+        String parsedRemoteId = mRemoteId.substring(0, Math.min(mRemoteId.length(), 8));
```

Comment:
The point here is that I get the complete remoteId from the server but I only need the first 8 characters.

Example: 

remoteId from server: **00000003ocr2n5bhxjuy**

-  **00000003** => file id
-  ocr2n5bhxjuy  => server instance ID

BTW, Math.min(mRemoteId.length() is not required, I will delete it